src/image.c

```diff
@@ -1614,9 +1621,16 @@ gdip_rotate_orthogonal_flip_x (GpImage *image, int angle, BOOL flip_x)
 	}
 
 	image->active_bitmap->scan0 = rotated;
-	image->active_bitmap->reserved |= GBD_OWN_SCAN0;	
-
-	gdip_bitmap_invalidate_surface (image);
+	image->active_bitmap->reserved |= GBD_OWN_SCAN0;
+
+    if (isSurfaceSource == 0)
+        gdip_bitmap_invalidate_surface(image);
+    else {
+        /* recalculate surface from rotated scan0 */
+        cairo_surface_destroy(image->surface);
+        image->surface = NULL;
+        gdip_bitmap_ensure_surface(image);
+    }
```

Comment:
please use tabs for indentation to follow the code style in this file.

Reply:
done, commit b72439c2c45eb3f0ebbe62b307330a8796014373